### PR TITLE
Split Modifiers to resolve flag clashes

### LINKED ISSUE #630: classes generated from Bytecode have synchronized modifier
According to https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html a class cannot be synchronised but it is set in Jimple files. I suggest that it sets the synchronized modifier because it is the same value as ACC_SUPER (0x0020) which can be an access flag of classes.

### PR DESCRIPTION
**Pull request by JonasKlauke**
Fields, Classes and Methods have modfiers which have the same bytecode but different meanings
fixes #630 
introduces API changes since Modifier is split to different Enums

### CODE REVIEW

File: sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java [hunk 1]
```diff
@@ -196,12 +196,39 @@ public Boolean visitFile(@Nonnull JimpleParser.FileContext ctx) {
...
+      Set<ClassModifier> modifierSet =
           modifier.stream()
-              .map(modifierContext -> Modifier.valueOf(modifierContext.getText().toUpperCase()))
+              .map(
+                  modifierContext -> ClassModifier.valueOf(modifierContext.getText().toUpperCase()))
               .collect(Collectors.toSet());
-      return modifierSet.isEmpty() ? EnumSet.noneOf(Modifier.class) : EnumSet.copyOf(modifierSet);
+      return modifierSet.isEmpty()
+          ? EnumSet.noneOf(ClassModifier.class)
+          : EnumSet.copyOf(modifierSet);
```
Comment (swissiety) [inline sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java]:
``` 
 .collect(Collectors.toCollection(() -> EnumSet.noneOf(ClassModifier.class)));
```

File: sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java [hunk 2]
```diff
@@ -196,12 +196,39 @@ public Boolean visitFile(@Nonnull JimpleParser.FileContext ctx) {
...
+
+    private EnumSet<MethodModifier> getMethodModifiers(
+        List<JimpleParser.ModifierContext> modifier) {
+      Set<MethodModifier> modifierSet =
+          modifier.stream()
+              .map(
+                  modifierContext ->
+                      MethodModifier.valueOf(modifierContext.getText().toUpperCase()))
+              .collect(Collectors.toSet());
+      return modifierSet.isEmpty()
```
Comment (swissiety) [inline sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java]: see above

File: sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java [hunk 3]
```diff
@@ -196,12 +196,39 @@ public Boolean visitFile(@Nonnull JimpleParser.FileContext ctx) {
...
+
+    private EnumSet<FieldModifier> getFieldModifiers(List<JimpleParser.ModifierContext> modifier) {
+      Set<FieldModifier> modifierSet =
+          modifier.stream()
+              .map(
+                  modifierContext -> FieldModifier.valueOf(modifierContext.getText().toUpperCase()))
+              .collect(Collectors.toSet());
+      return modifierSet.isEmpty()
+          ? EnumSet.noneOf(FieldModifier.class)
+          : EnumSet.copyOf(modifierSet);
```
Comment (swissiety) [inline sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java]: see above

**Review by swissiety — CHANGES_REQUESTED** [review]

File: sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java (hunk 1 above)
Comment (JonasKlauke) [reply sootup.jimple.parser/src/main/java/sootup/jimple/parser/JimpleConverter.java]: done

**Review by swissiety — APPROVED** [review]

**Pull request merged**